dmscripts/upload_counterpart_agreements.py

```diff
@@ -47,10 +70,23 @@ def upload_counterpart_file(bucket, framework_slug, file_path, dry_run, client):
             logger.info("[Dry-run] countersignedAgreementPath='{}' for agreement ID {}".format(
                 upload_path, supplier_framework['agreementId'])
             )
+
+        for notify_email in (notify_emails or ()):
+            if not dry_run:
```

Comment:
If we moved the notify client generation inside the `if dry_run` bit we wouldn't need this, but would probably want to leave it here for clarity anyway I guess.